api/gateway-operator/v2alpha1/controlplane_types.go

```diff
@@ -0,0 +1,236 @@
+type ControlPlaneOptions struct {
+	// DataPlane designates the target data plane to configure.
+	//
+	// It can be either a URL to an externally managed DataPlane (e.g. installed
+	// independently with Helm) or a name of a DataPlane resource that is
+	// managed by the operator.
+	//
+	// +required
+	DataPlane ControlPlaneDataPlaneTarget `json:"dataplane,omitempty"`
+
+	// Extensions provide additional or replacement features for the ControlPlane
+	// resources to influence or enhance functionality.
+	//
+	// +optional
+	// +kubebuilder:validation:MinItems=0
+	// +kubebuilder:validation:MaxItems=2
+	Extensions []commonv1alpha1.ExtensionRef `json:"extensions,omitempty"`
+
+	// WatchNamespaces indicates the namespaces to watch for resources.
+	//
+	// +optional
+	// +kubebuilder:default={type: all}
+	WatchNamespaces *operatorv1beta1.WatchNamespaces `json:"watchNamespaces,omitempty"`
+
+	// FeatureGates is a list of feature gates that are enabled for this ControlPlane.
+	//
+	// +optional
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MaxItems=32
+	FeatureGates []ControlPlaneFeatureGate `json:"featureGates,omitempty"`
+
+	// Controllers defines the controllers that are enabled for this ControlPlane.
+	//
+	// +optional
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MaxItems=32
+	Controllers []ControlPlaneController `json:"controllers,omitempty"`
+
+	// AdminAPI defines the configuration for the Kong Admin API.
+	//
+	// +optional
+	AdminAPI *ControlPlaneAdminAPI `json:"adminAPI,omitempty"`
+}
+
+// ControlPlaneDataPlaneTarget defines the target for the DataPlane that the ControlPlane
+// is responsible for configuring.
+//
+// +kubebuilder:validation:XValidation:message="URL has to be provided when type is set to url",rule="self.type != 'url' || has(self.url)"
+// +kubebuilder:validation:XValidation:message="Name cannot be provided when type is set to url",rule="self.type != 'url' || !has(self.name)"
+// +kubebuilder:validation:XValidation:message="Name has to be provided when type is set to name",rule="self.type != 'name' || has(self.name)"
+// +kubebuilder:validation:XValidation:message="URL cannot be provided when type is set to name",rule="self.type != 'name' || !has(self.url)"
+type ControlPlaneDataPlaneTarget struct {
+	// Type indicates the type of the DataPlane target.
+	//
+	// +kubebuilder:validation:Enum=url;name
+	// +kubebuilder:validation:Required
+	Type string `json:"type,omitempty"`
+
+	// URL is the URL of the DataPlane target. This is used for configuring
+	// externally managed DataPlanes like those installed independently with Helm.
+	//
+	// +optional
+	URL string `json:"url,omitempty"`
+
+	// Name is the name of the DataPlane to configure.
+	//
+	// +optional
+	Name string `json:"name,omitempty"`
+}
+
+// ControlPlaneAdminAPI defines the configuration for the Kong Admin API that
+// a ControlPlane when configuring the DataPlane.
```

Comment:
I guess we missed some words here?

Reply:
Yup, f287f8b87209a0db2fecfcbb64ab9a5530ee3036 should fix that.